navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt

```diff
@@ -23,47 +26,31 @@ import kotlinx.coroutines.flow.collectLatest
 @Composable
 fun BottomAppNavigation(
     ///Bottom navigation composable function that takes a
-    navigator: Navigator,
 
     ///Define the homeScreen, documentScreen, and searchScreen composable functions as parameters.
     homeScreen: @Composable () -> Unit,
     documentScreen: @Composable () -> Unit,
     searchScreen: @Composable () -> Unit
 ) {
-    val navController = rememberNavController()
 
-    ///Define a rememberDestination mutableState variable to store the current destination.
     val rememberDestination = remember { mutableStateOf(Destination.home.route) }
 
-    LaunchedEffect(Unit) {
-        navigator.actions.collectLatest {
-            when (it) {
-                is Navigator.NavigationActions.Navigate -> {
-                    navController.navigate(it.destination, builder = it.navOptions)
-                }
-
-                is Navigator.NavigationActions.Back -> {
-                    navController.popBackStack()
-                }
-            }
-        }
-    }
+    val state = rememberPagerState(initialPage = 0, pageCount = { bottomNavigationTabs.size })
 
     BottomNavigationWrapper(
         currentDestination = rememberDestination.value,
         body = {
-            NavHost(
-                navController = navController,
-                startDestination = Destination.home.route
-            ) {
-                composable(Destination.home.route) { homeScreen() }
-                composable(Destination.document.route) { documentScreen() }
-                composable(Destination.search.route) { searchScreen() }
+            HorizontalPager(state = state) { page ->
+                when (page) {
+                    0 -> homeScreen()
+                    1 -> documentScreen()
+                    2 -> searchScreen()
+                }
             }
         },
-        onChangeTab = { destination ->
-            rememberDestination.value = destination
-            navController.navigate(destination)
+        onChangeTab = { route, pageIndex ->
+            rememberDestination.value = route
+            state.requestScrollToPage(pageIndex)
```

Comment:
Directly using `state.requestScrollToPage(pageIndex)` may cause inconsistency issues. It's better to check if the requested page is different from the current page before making a call.

```diff
-            state.requestScrollToPage(pageIndex)
+            if (state.currentPage != pageIndex) {
+                state.requestScrollToPage(pageIndex)
+            }
```